validation/src/ma5_validation/utils/script_handler.py

```diff
@@ -0,0 +1,259 @@
+
+from ma5_validation.system.exceptions import InvalidScript
+from .mode_handler import MA5Mode
+from .path_handler import PathHandler
+
+
+class ScriptReader:
+    """
+    Validation script reader.
+
+    Format of the valiation script:
+    ```
+    #TITLE Script title
+    #MODE Ma5 run mode
+
+    plot MET
+    select MET > 20
+    import $MA5PATH/samples/*lhe*
+    ```
+    `$MA5PATH` indicates madanalysis path which has been in use with the current validation run.
+
+    Parameters
+    ----------
+    filename : Text
+        script name or full path.
+    """
+
+    _modes = MA5Mode._member_names_
+
+    def __init__(self, filename: Text = None, name: Text = None):
+        if name is None:
+            self.name = os.path.basename(filename).split(".ma5")[0]
+        else:
+            self.name = name
+        self.filename = filename
+        self._mode = None
+        self.title = None
+        self._ma5_commands = []
+
+    @staticmethod
+    def _modifier(line: Text) -> Text:
+        """
+        Modifies the input string's keywords
+
+        Parameters
+        ----------
+        line : Text
+            input string
+
+        Returns
+        -------
+        Text:
+            modified line
+        """
+        modifications = {
+            "$MA5PATH": PathHandler.MA5PATH,
+            "$SMP_PATH": PathHandler.SMP_PATH,
+            "$LOGPATH": PathHandler.LOGPATH,
+            "$PARTON_LEVEL_PATH": PathHandler.PARTON_LEVEL_PATH,
+            "$HADRON_LEVEL_PATH": PathHandler.HADRON_LEVEL_PATH,
+            "$RECO_LEVEL_PATH": PathHandler.RECO_LEVEL_PATH,
+            "$EXPERT_LEVEL_PATH": PathHandler.EXPERT_LEVEL_PATH,
+            "$FASTJET_INTERFACE_PATH": PathHandler.FASTJET_INTERFACE_PATH,
+            "$DELPHES_INTERFACE_PATH": PathHandler.DELPHES_INTERFACE_PATH,
+        }
+        for key, path in modifications.items():
+            line = line.replace(key, path)
+        return line
+
+    def decode(self) -> None:
+        """
+        Decode MadAnalysis 5 validation script. Format:
+
+        ```
+        #TITLE Script title
+        #MODE Ma5 run mode
+        #SUBMIT Submission folder name i.e. sfs_test
+
+        plot MET
+        select MET > 20
+        ```
+        """
+        if self.filename is None:
+            raise InvalidScript("File name is not defined.")
+
+        if len(self.filename.split("/")) > 1:
+            if os.path.isfile(self.filename):
+                filename = self.filename
+            else:
+                raise FileNotFoundError(f"Can't find ma5 script: {self.filename}")
+        else:
+            filename = os.path.join(
+                os.path.split(os.path.dirname(os.path.realpath(__file__)))[0],
+                "ma5_scripts",
+                self.filename,
+            )
+            if not os.path.isfile(filename):
+                raise FileNotFoundError(f"Can't find ma5 script: {filename}")
+
+        with open(filename, "r", encoding="utf-8") as script:
```

Comment:
@econte-cms In the `decode` function `script_handler` checks all the keywords, so we can expand the keyword structure to accommodate numerical comparisons. Also, all paths are taken care of by `path_handler` so we don't need to declare complete paths in the script file, path keywords are replaced by the relative path with respect to the system.